lib/Sema/CSWalker.cpp

```diff
@@ -283,6 +264,124 @@ class LocalCSWalker : public glu::ast::ASTWalker<LocalCSWalker, void> {
         );
         _cs.addConstraint(constraint);
     }
+
+    void postVisitCallExpr(glu::ast::CallExpr *node)
+    {
+        auto *refExpr = llvm::dyn_cast<glu::ast::RefExpr>(node->getCallee());
+        if (!refExpr)
+            return handlePointerCall(node);
+
+        llvm::StringRef functionName = refExpr->getIdentifier();
+
+        llvm::SmallVector<glu::types::TypeBase *, 4> argTypes;
+        for (auto *arg : node->getArgs()) {
+            argTypes.push_back(arg->getType());
+        }
+
+        auto &arena = node->getModule()->getContext()->getTypesMemoryArena();
+        auto *expectedFnTy
+            = arena.create<glu::types::FunctionTy>(argTypes, node->getType());
+
+        bool success = resolveOverloadSet(
+            functionName, node, expectedFnTy,
+            /*isExplicit=*/true
+        );
+
+        if (!success) {
+            _diagManager.error(
+                node->getLocation(),
+                "No function overloads found for '" + functionName.str() + "'"
+            );
+        }
+    }
+
+private:
+    /// @brief Resolve overload set for a given name, creating a disjunction
+    /// constraint.
+    /// @param name The identifier (operator or function name).
+    /// @param anchor The AST node to attach the constraint and diagnostics to.
+    /// @param expectedFnTy The expected function type, if any.
+    /// @param isExplicit Whether this is an explicit call (e.g. CallExpr).
+    /// @returns True if any overload constraints were added; false if none
+    /// matched.
+    bool resolveOverloadSet(
+        llvm::StringRef name, glu::ast::ExprBase *anchor,
+        glu::types::FunctionTy *expectedFnTy, bool isExplicit
+    )
+    {
+        auto *item = _cs.getScopeTable()->lookupItem(name);
+        llvm::ArrayRef<glu::ast::DeclBase *> decls
+            = item ? item->decls : llvm::ArrayRef<glu::ast::DeclBase *>();
+
+        llvm::SmallVector<Constraint *, 4> constraints;
+
+        for (auto *decl : decls) {
+            if (auto *fnDecl = llvm::dyn_cast<glu::ast::FunctionDecl>(decl)) {
+                constraints.push_back(
+                    Constraint::createConjunction(
+                        _cs.getAllocator(),
+                        { Constraint::createBindOverload(
+                              _cs.getAllocator(), anchor->getType(), fnDecl,
+                              anchor
```

Comment:
The node associated with a BindOverload must be a RefExpr, which is why I added them to operators